safehttp/plugins/xsrf/xsrf_test.go

```diff
@@ -163,87 +91,204 @@ func TestXSRFTokenMultipart(t *testing.T) {
 			wantBody: "Forbidden\n",
 		},
 	}
-	for _, test := range tests {
-		i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
-		tok := xsrftoken.Generate("xsrf", test.userID, test.host+test.path)
-		b := "--123\r\n" +
-			"Content-Disposition: form-data; name=\"xsrf-token\"\r\n" +
-			"\r\n" +
-			tok + "\r\n" +
-			"--123--\r\n"
-		req := safehttptest.NewRequest("POST", "http://foo.com/pizza", strings.NewReader(b))
-		req.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
+)
 
-		rec := safehttptest.NewResponseRecorder()
-		i.Before(rec.ResponseWriter, req)
+func TestTokenPost(t *testing.T) {
+	for _, method := range stateChangingMethods {
+		for _, test := range formTokenTests {
+			t.Run(method+test.name, func(t *testing.T) {
+				i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
+				tok := xsrftoken.Generate("xsrf", test.userID, test.actionID)
+				req := safehttptest.NewRequest(method, "https://foo.com/pizza", strings.NewReader(xsrf.TokenKey+"="+tok))
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+				rec := safehttptest.NewResponseRecorder()
 
-		if rec.Status() != test.wantStatus {
-			t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
-		}
-		if diff := cmp.Diff(test.wantHeader, map[string][]string(rec.Header())); diff != "" {
-			t.Errorf("rw.header mismatch (-want +got):\n%s", diff)
+				i.Before(rec.ResponseWriter, req)
+
+				if rec.Status() != test.wantStatus {
+					t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
+				}
+				if diff := cmp.Diff(test.wantHeader, map[string][]string(rec.Header())); diff != "" {
+					t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+				}
+				if got := rec.Body(); got != test.wantBody {
+					t.Errorf("response body: got %q want %q", got, test.wantBody)
```

Comment:
This error message could start with `"rec.Body(): ..."`

Reply:
We should make sure all our tests use either `response status` and `response body` or the function names.

Reply:
Agreed, and I think it should be the function names

Reply:
Opened #126